### PR TITLE
Stats revamp v2:  Remove feature config

### LINKED ISSUE #16988: [android] Remove feature config

### PR DESCRIPTION
**Pull request by ravishanker**
This PR remove **StatsRevampV2 Feature Config**, **Stats New Feature Intro**  and updates all the places that was in use including analytics and prefs.

Fixes #16988 

To test:

- Build and run
- Launch **Jetpack app**
- Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Select Debug Settings
- Ensure **StatsRevampV2FeatureConfig** is not present
- Ensure no Stats Feature Intro is shown
- Navigate to Stats, and ensure the new cards are shown or a pop-up is displayed about update and new cards shown
- Also, launch **WordPress app** and ensure it is still showing old stats cards on Stats screen.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
updated unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/build.gradle [hunk 1]
```diff
@@ -109,7 +109,8 @@ android {
         buildConfigField "boolean", "SITE_NAME", "false"
         buildConfigField "boolean", "LAND_ON_THE_EDITOR", "false"
         buildConfigField "boolean", "BLOGGING_PROMPTS", "false"
-        buildConfigField "boolean", "STATS_REVAMP_V2", "false"
+        buildConfigField "boolean", "MY_SITE_DEFAULT_TAB_EXPERIMENT", "false"
+        buildConfigField "boolean", "MY_SITE_DEFAULT_TAB_EXPERIMENT_VARIANT_DASHBOARD", "false"
```
Comment (irfano) [inline WordPress/build.gradle]: `MY_SITE_DEFAULT_TAB_EXPERIMENT` and `MY_SITE_DEFAULT_TAB_EXPERIMENT_VARIANT_DASHBOARD` don't look related to this PR. Did they come from trunk merge?

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt [hunk 2]
```diff
@@ -109,7 +108,7 @@ class StatsModule {
     @Named(BLOCK_INSIGHTS_USE_CASES)
     @Suppress("LongParameterList")
     fun provideBlockInsightsUseCases(
-        statsRevampV2FeatureConfig: StatsRevampV2FeatureConfig,
+//        statsRevampV2FeatureConfig: StatsRevampV2FeatureConfig,
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt]: np: I'd prefer removing this instead commenting.

**Review by irfano — CHANGES_REQUESTED** [review]
My manual test was successful, but some unit tests fail after removing the flag.

File: WordPress/build.gradle (hunk 1 above)
Comment (ravishanker) [reply WordPress/build.gradle]: Yes

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt (hunk 2 above)
Comment (ravishanker) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt]: Thanks for spotting it.  Cleaned-up now.

**Comment by ravishanker** [comment]
> My manual test was successful, but some unit tests fail after removing the flag.

I did update unit tests.  They ran successfully locally and also the build as you can see is also green.  What unit tests are failing for you and why?

**Comment by irfano** [comment]
> I did update unit tests. They ran successfully locally and also the build as you can see is also green. What unit tests are failing for you and why?

`InsightsManagementMapperTest` was the failing test. But they are working now. Maybe it was a cache problem on my local.

**Review by irfano — APPROVED** [review]
Removing feature config change LGTM! 👍🏻 
I'm approving, but I'm not sure about added `MY_SITE_DEFAULT_TAB_EXPERIMENT` and `MY_SITE_DEFAULT_TAB_EXPERIMENT_VARIANT_DASHBOARD`. They came from the trunk merge, but they should have not been in the diff of the PR. The current trunk doesn't contain `MY_SITE_DEFAULT_TAB_EXPERIMENT`.
I think we should remove them from this PR. WDYT?

**Comment by ravishanker** [comment]
> Removing feature config change LGTM! 👍🏻 I'm approving, but I'm not sure about added `MY_SITE_DEFAULT_TAB_EXPERIMENT` and `MY_SITE_DEFAULT_TAB_EXPERIMENT_VARIANT_DASHBOARD`. They came from the trunk merge, but they should have not been in the diff of the PR. The current trunk doesn't contain `MY_SITE_DEFAULT_TAB_EXPERIMENT`. I think we should remove them from this PR. WDYT?

Both those lines have been removed

**Pull request merged**